zha/application/platforms/light/__init__.py

```diff
@@ -1044,8 +1045,8 @@ def _assume_group_state(self, update_params) -> None:
             if hs_color is not None and ColorMode.HS in supported_modes:
                 self._hs_color = hs_color
             # the effect is always deactivated in async_turn_on if not provided
-            if effect is None:
-                self._effect = None
+            if effect is None or effect == EFFECT_OFF:
+                self._effect = EFFECT_OFF
             elif self._effect_list and effect in self._effect_list:
                 self._effect = effect
```

Comment:
I believe this still needs some changes, though I haven't tested the PR yet.
This might be what we actually want here:
```suggestion
            if effect is not None and effect in self._effect_list:
                self._effect = effect
```
(It's `None` if the `effect` wasn't touched on the group. In that case, we do not want to modify the member lights. If it's not None and in the effect list, the member will have applied that effect, so we want to set that state too. This will include `EFFECT_OFF`.)